Use length framing instead of delimiter for JSON over socket

aldafu opened the pull request:
For each JSON string serialized message object to send, use a Buffer
prefixed with four octets for an unsigned int (little endian) that
specifies the length of the serialized message.

http://jira.webinos.org/browse/WP-49

aldafu committed:
Use length framing instead of delimiter for JSON over socket

For each JSON string serialized message object to send, use a Buffer
prefixed with four octets for an unsigned int (little endian) that
specifies the length of the serialized message.

Fixes Jira-issue: WP-49

ghost commented:
Due to Habib's commits to make webinos standalone, this commit can not be merged automatically. I tried to merge manually, but there are a large number of conflicts. Could you try to rebase your branch so it is replayed on top of current master. The first section of this page http://git-scm.com/book/en/Git-Branching-Rebasing should explain what I mean.
First you need to check out this specific branch and then issue git rebase master. There will be conflicts, which need to be resolved.
Thanks!

aldafu committed:
Fix debug message call.

Jira-issue: WP-49

aldafu committed:
Fix log calls with typod error type. Again.

The wrong type for these log output calls resulted in errors not being
logged.

Jira-issue: WP-49

aldafu commented:
Okay, rebased it to current master. Should apply cleanly now.

ghost commented:
Indeed, thanks!
